addons/pos_restaurant/static/src/js/Screens/ProductScreen/ControlButtons/SplitBillButton.js

```diff
@@ -14,7 +14,7 @@ odoo.define('pos_restaurant.SplitBillButton', function(require) {
         async onClick() {
             const order = this.env.pos.get_order();
             if (order.get_orderlines().length > 0) {
-                this.showScreen('SplitBillScreen');
+                this.showScreen('SplitBillScreen', { disallow: this.__owl__.parent.disallowLineQuantityChange() });
```

Comment:
I don't really like this thing, as it is used in productScreen and in BillScreen, wouldn't it be more elegant to set it in the environement? it'll also let this information accessible, if we need to restrict other things in other screens...

@caburj @rhe-odoo What do you think?

Reply:
It's what I suggested him but I don't like it either. Perhaps he can copy the method from productscreen as a temporary solution. I'll think of better way.

Reply:
Actually, it's possible to lift the `disallowLineQuantityChange` to the PosModel instance (pos). To do it, we need to also put `numpadMode` to pos. It looks like we are dumping stuff to pos but IMO, it's not that bad.

Reply:
@rhe-odoo So yeah, I guess @pimodoo and I agree with putting the value to the environment. I believe you can do this:

```js
// models.js
class PosModel {
  ...
  disallowLineQuantityChange() {
    let result = super.disallowLineQuantityChange();
    return (this.is_french_country() && this.numpadMode === 'quantity') || result;
  }
}

class ProductScreen {
  ...
  _setNumpadMode(mode) {
    ...
    this.env.pos.numpadMode = mode;
  }
}
```